From: Kai Uwe Broulik <kde@privat.broulik.de>
Date: Sat, 11 Jul 2015 00:06:52 +0000
Subject: Clear view lists in destructor

This fixes a crash on shutdown when it has already deleted its
views but someone calls into availableScreenRect

Origin: upstream, http://quickgit.kde.org/?p=plasma-workspace.git&a=commitdiff&h=4942878b67c6a5ae82df11264eb751944cea4a69
Bug: https://bugs.kde.org/show_bug.cgi?id=348511
Bug-Debian: http://bugs.debian.org/793513

--- a/shell/shellcorona.cpp
+++ b/shell/shellcorona.cpp
@@ -196,8 +196,10 @@
 ShellCorona::~ShellCorona()
 {
     qDeleteAll(m_views);
+    m_views.clear();
     qDeleteAll(containments());
     qDeleteAll(m_panelViews);
+    m_panelViews.clear();
 }
 
 KPackage::Package ShellCorona::lookAndFeelPackage()

